code/modules/clothing/head/soft_caps.dm

```diff
@@ -161,3 +162,40 @@
 /obj/item/clothing/head/soft/fishing_hat/Initialize(mapload)
 	. = ..()
 	AddElement(/datum/element/skill_reward, /datum/skill/fishing)
+
+/obj/item/clothing/head/soft/propeller_hat
+	name = "propeller hat"
+	desc = "A colorful hat with a spinning propeller sat on top."
+	icon_state = "propeller_hat"
+	soft_type = "propeller_hat"
+	inhand_icon_state = "rainbow_softcap"
+	worn_y_offset = 1
+	soft_suffix = null
+	actions_types = list(/datum/action/item_action/toggle)
+	var/enabled_waddle = TRUE
+	var/active = FALSE
+
+/obj/item/clothing/head/soft/propeller_hat/update_icon_state()
+	. = ..()
+	worn_icon_state = "[soft_type][flipped ? "_flipped" : null][active ? "_on" : null]"
+
+/obj/item/clothing/head/soft/propeller_hat/attack_self(mob/user)
+	active = !active
+	to_chat(user, span_notice("You [active ? "start" : "stop"] spinning the propeller."))
```

Comment:
If this has an obvious animation, then you don't need any sort of notice

Reply:
Ok looks like it doesn't affect the actual icon state other than the worn. In that case, just a balloon alert is necessary.

```suggestion
	balloon_alert(user, span_notice(active ? "started propeller" : "stopped propeller"))
```